gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystem.java

```diff
@@ -589,6 +589,24 @@ public FSDataInputStream open(Path hadoopPath, int bufferSize) throws IOExceptio
         });
   }
 
+  public FSDataInputStream open(FileStatus status) throws IOException {
+    logger.atFine().log("openWithStatus(%s)", status);
+
+    GoogleHadoopFileStatus fileStatus = (GoogleHadoopFileStatus) status;
+
+    return trackDurationWithTracing(
+        instrumentation,
+        globalStorageStatistics,
+        GhfsStatistic.INVOCATION_OPEN,
+        status.getPath(),
+        this.traceFactory,
+        () -> {
+          checkOpen();
+          return new FSDataInputStream(
+              GoogleHadoopFSInputStream.create(this, fileStatus.getFileInfo(), statistics));
```

Comment:
IIUC, there is no restriction on the bucket of provided FileStatus to the bucket with which ghfs is initialized. Is it an intended behaviour? Isn't it deviating from all other APIs?

Reply:
Thank you for the suggestion. Updated.